tests/drivers/watchdog/wdt_basic_api/testcase.yaml

```diff
@@ -1,21 +1,20 @@
-tests:
-  drivers.watchdog:
+common:
     depends_on: watchdog
     tags: drivers watchdog
-    filter: not (CONFIG_WDT_SAM or dt_compat_enabled("st,stm32-window-watchdog"))
+tests:
+  drivers.watchdog:
+    filter: not (CONFIG_WDT_SAM or dt_compat_enabled("st,stm32-window-watchdog") or dt_compat_enabled("st,stm32-watchdog"))
     platform_exclude: mec15xxevb_assy6853 mimxrt685_evk_cm33
   drivers.watchdog.stm32wwdg:
-    depends_on: watchdog
-    tags: drivers watchdog
-    filter: dt_compat_enabled("st,stm32-window-watchdog")
+    filter: dt_compat_enabled("st,stm32-window-watchdog") or dt_compat_enabled("st,stm32-window-watchdog")
+    extra_args: DTC_OVERLAY_FILE="boards/stm32_wwdg.overlay"
```

Comment:
This filter seems totally fine to me and yet, it doesn't have the expected effect, as twister still atempts to build it on non complying boards:
```
INFO    - 130/584 frdm_k22f                 tests/drivers/watchdog/wdt_basic_api/drivers.watchdog.stm32iwdg FAILED Cmake build failure (build)
INFO    - /__w/zephyr/zephyr/twister-out/frdm_k22f/tests/drivers/watchdog/wdt_basic_api/drivers.watchdog.stm32iwdg/build.log
ERROR   - Loading Zephyr default modules (Zephyr base).
-- Application: /__w/zephyr/zephyr/tests/drivers/watchdog/wdt_basic_api
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Cache files will be written to: /__w/zephyr/zephyr/.cache
-- Zephyr version: 3.1.0-rc1 (/__w/zephyr/zephyr)
-- Found west (found suitable version "0.13.0", minimum required is "0.7.1")
-- Board: frdm_k22f
-- Found host-tools: zephyr 0.14.1 (/opt/toolchains/zephyr-sdk-0.14.1)
-- Found dtc: /opt/toolchains/zephyr-sdk-0.14.1/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found toolchain: zephyr 0.14.1 (/opt/toolchains/zephyr-sdk-0.14.1)
-- Found BOARD.dts: /__w/zephyr/zephyr/boards/arm/frdm_k22f/frdm_k22f.dts
-- Found devicetree overlay: boards/stm32_iwdg.overlay
devicetree error: ./boards/stm32_iwdg.overlay:7 (column 1): parse error: undefined node label 'wwdg'
```